core/primitives/Cargo.toml

```diff
@@ -89,7 +89,7 @@ nightly_protocol = [
   "near-primitives/nightly_protocol",
 ]
 
-default = ["near-primitives/rand"]
+default = ["rand", "clock"]
```

Comment:
Does `clock` really need to be enabled by default?

It pulls in `tokio` as a dependency:

https://github.com/near/nearcore/blob/1eeee00448cff71c1f6b994f63fae009f9dfc788/core/time/Cargo.toml#L14-L24

Reply:
not sure if even rand is mandatory ...

https://github.com/near/nearcore/commit/04d2b4f7e3c2882f8873f52bd6ee5355eb5b298b this PR introduces default-less

Reply:
I am open to suggestions. So far, if I remove `clock`, `cargo ws exec -- cargo check` will fail on
```
Checking near-primitives v0.0.0                                                                                                                             
error[E0433]: failed to resolve: could not find `serde_utc_as_iso` in `near_time`                                                                                                                                   
   --> core/primitives/src/views.rs:337:20                                                                                                                                                                          
    |                                                                                                                                                                                                               
337 |     #[serde(with = "near_time::serde_utc_as_iso")]                                                                                                                                                            
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `serde_utc_as_iso` in `near_time`                                                                                                             
                                                                                                               
```
If I add "near-time/serde" then it fails on 
```
--> core/dyn-configs/src/lib.rs:6:5
  |
6 | use near_time::Clock;
  |     ^^^^^^^^^^^^^^^^ no `Clock` in the root
```
so we need both deps that are included under `clock`.

Reply:
probably, somebody added without feature-gates :(

we need to restrict it futher

Reply:
I added another commit.

Reply:
```diff
diff --git a/core/dyn-configs/Cargo.toml b/core/dyn-configs/Cargo.toml
index 15f68df4d..ccbffb4ec 100644
--- a/core/dyn-configs/Cargo.toml
+++ b/core/dyn-configs/Cargo.toml
@@ -20,7 +20,7 @@ thiserror.workspace = true
 tokio.workspace = true
 tracing.workspace = true
 
-near-time.workspace = true
+near-time = { workspace = true, features = ["clock"] }
 near-chain-configs.workspace = true
 near-crypto.workspace = true
 near-o11y.workspace = true
diff --git a/core/primitives/Cargo.toml b/core/primitives/Cargo.toml
index d592da655..a7cfdbbb2 100644
--- a/core/primitives/Cargo.toml
+++ b/core/primitives/Cargo.toml
@@ -58,43 +58,40 @@ rand = ["dep:rand", "rand_chacha", "near-crypto/rand", "itertools"]
 clock = ["near-time/clock", "near-time/serde"]
 dump_errors_schema = ["near-rpc-error-macro/dump_errors_schema"]
 protocol_feature_fix_staking_threshold = [
-  "near-primitives-core/protocol_feature_fix_staking_threshold",
+    "near-primitives-core/protocol_feature_fix_staking_threshold",
 ]
 protocol_feature_fix_contract_loading_cost = [
-  "near-primitives-core/protocol_feature_fix_contract_loading_cost",
+    "near-primitives-core/protocol_feature_fix_contract_loading_cost",
 ]
 protocol_feature_reject_blocks_with_outdated_protocol_version = [
-  "near-primitives-core/protocol_feature_reject_blocks_with_outdated_protocol_version",
+    "near-primitives-core/protocol_feature_reject_blocks_with_outdated_protocol_version",
 ]
 protocol_feature_nonrefundable_transfer_nep491 = [
-  "near-primitives-core/protocol_feature_nonrefundable_transfer_nep491",
+    "near-primitives-core/protocol_feature_nonrefundable_transfer_nep491",
 ]
 
 nightly = [
-  "near-fmt/nightly",
-  "near-parameters/nightly",
-  "near-primitives-core/nightly",
-  "near-primitives/nightly",
-  "nightly_protocol",
-  "protocol_feature_fix_contract_loading_cost",
-  "protocol_feature_fix_staking_threshold",
-  "protocol_feature_nonrefundable_transfer_nep491",
-  "protocol_feature_reject_blocks_with_outdated_protocol_version",
+    "near-fmt/nightly",
+    "near-parameters/nightly",
+    "near-primitives-core/nightly",
+    "near-primitives/nightly",
+    "nightly_protocol",
+    "protocol_feature_fix_contract_loading_cost",
+    "protocol_feature_fix_staking_threshold",
+    "protocol_feature_nonrefundable_transfer_nep491",
+    "protocol_feature_reject_blocks_with_outdated_protocol_version",
 ]
 
 nightly_protocol = [
-  "near-fmt/nightly_protocol",
-  "near-parameters/nightly_protocol",
-  "near-primitives-core/nightly_protocol",
-  "near-primitives/nightly_protocol",
+    "near-fmt/nightly_protocol",
+    "near-parameters/nightly_protocol",
+    "near-primitives-core/nightly_protocol",
+    "near-primitives/nightly_protocol",
 ]
 
-default = ["rand", "clock"]
 calimero_zero_storage = []
 
-protocol_schema = [
-  "near-schema-checker-lib/protocol_schema",
-]
+protocol_schema = ["near-schema-checker-lib/protocol_schema"]
 
 [dev-dependencies]
 chrono = { workspace = true, features = ["clock"] }
diff --git a/core/time/Cargo.toml b/core/time/Cargo.toml
index 408a22b98..f874b2678 100644
--- a/core/time/Cargo.toml
+++ b/core/time/Cargo.toml
@@ -20,5 +20,6 @@ serde = { workspace = true, optional = true }
 serde_json.workspace = true
 
 [features]
+default = ["serde"]
 clock = ["tokio"]
 serde = ["dep:serde", "time/serde"]
```

I think, this diff should make it working

Reply:
You can apply it. Please, note thhat I have made on top of previous your commit

Reply:
It worked! Thanks for the input!